tree/ntupleutil/v7/src/RNTupleImporter.cxx

```diff
@@ -239,8 +239,10 @@ ROOT::Experimental::RResult<void> ROOT::Experimental::RNTupleImporter::PrepareSc
       ib.fBranchBuffer = std::make_unique<unsigned char[]>(branchBufferSize);
       if (isClass) {
          auto klass = TClass::GetClass(firstLeaf->GetTypeName());
-         if (!klass)
-            return R__FAIL("unable to load class " + std::string(firstLeaf->GetTypeName()));
+         if (!klass) {
+            return R__FAIL("unable to load class " + std::string(firstLeaf->GetTypeName()) + " for branch " +
```

Comment:
In which circumstance do you see this?  The ROOT files contains (or at least are supposed to contain) all the information needed to create a `TClass` for all the classes of objects stored in the file.  So even-though reaching this error "could" happen, it should not and is likely indicating a logic error earlier (I.e. you should not (in normal case) have both `(isClass == true)` and `(klass == nullptr)`

Reply:
Likely related:
```
   const bool isClass = (firstLeaf->IsA() == TLeafElement::Class()); // STL or user-defined class
```
is misleading.  The fact that the leaf is a `TLeafElement` 'only' means that the element is part a (split) object.  It's own type can be anything.

See `TTree::CheckBranchAddressType` and `TBranch::GetExpectedType` and `TBranchElement::GetExpectedType` for some of the way the underlying type can be found.

Reply:
Ah, thanks! That is most likely it. The error message we hit was on `Int_t`, i.e. there must be a branch of type `Int_t` and the current code thinks that it is a class.